src/app/frontend/deploy/createsecret_controller.js

```diff
@@ -0,0 +1,166 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import DeployLabel from './deploylabel';
+
+/**
+ * Secret creation dialog controller.
+ *
+ * @final
+ */
+export default class CreateSecretController {
+  /**
+   * @param {!md.$dialog} $mdDialog
+   * @param {!angular.$log} $log
+   * @param {!angular.$resource} $resource
+   * TODO (cheld) Set correct type after fixing issue #159
+   * @param {!Object} errorDialog
+   * @param {string} namespace
+   * @ngInject
+   */
+  constructor($mdDialog, $log, $resource, errorDialog, namespace) {
+    /** @private {!md.$dialog} */
+    this.mdDialog_ = $mdDialog;
+
+    /** @private {!angular.$log} */
+    this.log_ = $log;
+
+    /** @private {!angular.$resource} */
+    this.resource_ = $resource;
+
+    /**
+     * TODO (cheld) Set correct type after fixing issue #159
+     * @private {!Object}
+     */
+    this.errorDialog_ = errorDialog;
+
+    /** @export {!angular.FormController} */
+    this.secretForm;
+
+    /** @export {string} */
+    this.namespace = namespace;
+
+    /** @export {string} */
+    this.secretName;
+
+    /** @export {!Object<string, string>} */
+    this.data = {};
+
+    /**
+     * TODO: (taimir) Validate the keys of the key-value pairs
+     * @export {!Array<!DeployLabel>}
+     */
+    this.labels = [
+      new DeployLabel(),
+    ];
+
+    /**
+     * Max-length validation rule for secretName
+     * @export {number}
+     */
+    this.secretNameMaxLength = 253;
+
+    /**
+     * Pattern validation rule for secretName
+     * @export {RegExp}
+     */
+
+    this.secretNamePattern =
+        new RegExp('^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$');
+  }
+
+  /**
+   * Returns true if new secret name hasn't been filled by the user, i.e, is empty.
+   * @return {boolean}
+   * @export
+   */
+  isDisabled() { return (!this.secretName || !this.isKeyValueDataDefined_(this.labels)); }
+
+  /**
+   * Cancels the create secret form.
+   * @export
+   */
+  cancel() { this.mdDialog_.cancel(); }
+
+  /**
+   * Creates new secret based on the state of the controller.
+   * @export
+   */
+  createSecret() {
+    if (!this.secretForm.$valid) return;
+
+    /** @type {!backendApi.SecretSpec} */
+    let secretSpec = {
+      name: this.secretName,
+      namespace: this.namespace,
+      type: '',
+      data: this.toSecretData_(this.labels),
+    };
+
+    /** @type {!angular.Resource<!backendApi.SecretSpec>} */
+    let resource = this.resource_(`/api/secrets/`);
```

Comment:
Dont start api calls with `/`. Just `resource_('api/secrets')`. This is because the UI needs to work even when it is not attached to the root of the domain.